src/Lean/Elab/StructInst.lean

```diff
@@ -197,13 +193,15 @@ private def elabModifyOp (stx modifyOp : Syntax) (sources : Array ExplicitSource
 private def getStructName (expectedType? : Option Expr) (sourceView : Source) : TermElabM Name := do
   tryPostponeIfNoneOrMVar expectedType?
   let useSource : Unit → TermElabM Name := fun _ => do
-    match sourceView, expectedType? with
-    | .explicit sources, _ =>
-      if sources.size > 1 then
-        throwErrorAt sources[1]!.stx "invalid \{...} notation, expected type is not known, using the type of the first source, extra sources are not needed"
-      return sources[0]!.structName
-    | _, some expectedType => throwUnexpectedExpectedType expectedType
-    | _, none              => throwUnknownExpectedType
+    match sourceView.explicit.size with
+    | 0 =>
+      match expectedType? with
+      | some expectedType => throwUnexpectedExpectedType expectedType
+      | none => throwUnknownExpectedType
+    | 1 => return sourceView.explicit[0]!.structName
+    | _ =>
+      throwErrorAt sourceView.explicit[1]!.stx
+        "invalid \{...} notation, expected type is not known, using the type of the first source, extra sources are not needed"
```

Comment:
This error message seems to be incorrect by the way. It says "using the type of the first source" but it's using `throwErrorAt` which is a fatal error. Should it be using a non-fatal error here?

Reply:
I agree. We could just use the type of the first source. Did you try it?

Reply:
> Should it still be a non-fatal error? Or should we just not report anything

I am not sure. If we don't report anything I feat it may produce counterintuitive behavior and confuse users. What do you think?

Reply:
BTW, I am fine with merging as-is, and addressing this issue in another PR or just keeping it as-is.

Reply:
I'm currently compiling a fix to just report nothing and use the first source. I think it's teachable enough, and also fairly easy to override using the expected type syntax.